### ISSUE TITLE
ENH: allow pd.merge / join to create a MultiIndex result for columns

### ISSUE THREAD

**Issue by jreback** [pandas-dev/pandas#15060]
If no ``suffixes`` are specified, you get '_x', '_y' behavior.

```
In [76]: a = pd.DataFrame(dict(a=np.arange(5), b=np.arange(5), c=np.arange(5)))

In [77]: b = pd.DataFrame(dict(a=np.arange(6), b=[0,1,2,3,5,6]))

In [78]: a
Out[78]:
   a  b  c
0  0  0  0
1  1  1  1
2  2  2  2
3  3  3  3
4  4  4  4

In [79]: b
Out[79]:
   a  b
0  0  0
1  1  1
2  2  2
3  3  3
4  4  5
5  5  6

In [80]: res = pd.merge(a, b, left_index=True, right_index=True)

In [81]: res
Out[81]:
   a_x  b_x  c  a_y  b_y
0    0    0  0    0    0
1    1    1  1    1    1
2    2    2  2    2    2
3    3    3  3    3    3
4    4    4  4    4    5
```

If can be useful to allow ``suffix='expand'`` (or something like this). to enable an automatic
creation of a column multiindex (this is analagous to ``keys=`` arg in ``pd.concat``)

```
In [82]: res.columns = res.columns.str.split('_', expand=True)

In [83]: res = res.sort_index(axis=1)

In [84]: res
Out[84]:
   a     b      c
   x  y  x  y NaN
0  0  0  0  0   0
1  1  1  1  1   1
2  2  2  2  2   2
3  3  3  3  3   3
4  4  4  4  5   4
```

**Comment by hgilde** [comment]
IMO in this case, column c should get 'x' as level 1 to make everything consistent.

**Comment by jorisvandenbossche** [comment]
We have ``pd.concat`` for this, I would say? (and not further complicate `merge`)

**Comment by jreback** [comment]
yeah the issue is if you are actually merging (e.g. a unique & a non-unique merge column).

**Comment by hgilde** [comment]
agreed that concat has the feature, but doesn't handle as many cases as merge

**Comment by jorisvandenbossche** [comment]
The basic inner or outer join with unique and non-unique index still works with concat I think?

Also, if you want MultiIndex when suffixes `_x` and `_y` are added, you can easily get this with: `result.columns = result.columns.str.split('_', expand=True)`

**Comment by jreback** [comment]
@jorisvandenbossche I indicated this above. The point is to make this as an option to ``suffix=``.

**Comment by jorisvandenbossche** [comment]
@jreback Yep, I understand, I was just playing the advocate of the devil and mentioning reasons why this may not be worth the added complexity.

**Comment by jbrockmendel** [comment]
I'm unclear on the extent to which the concat alternative satisfies the original request.  If it does, then I completely agree with @jorisvandenbossche we don't need to make merge more complicated.
